.circleci/config.yml

```diff
@@ -0,0 +1,198 @@
+    docker:
+      - image: cirrusci/flutter:stable
+    steps:
+      - checkout
+      - build-flutter-project
+      - run:
+          name: Analyze code
+          command: flutter analyze lib
+
+  test:
+    description: "Run tests for Flutter"
+    docker:
+      - image: cirrusci/flutter:stable
+    steps:
+      - checkout
+      - build-flutter-project
+      - run:
+          name: Run tests
+          command: flutter test
+
+  android-integration-test-build:
+    description: "Build Android Flutter integration tests"
+    docker:
+      - image: cirrusci/flutter:stable
+    steps:
+      - checkout
+      - run:
+          name: Change api key
+          command: sed -i s/api_key/$API_KEY/ example/integration_test/app_test.dart
+      - run:
+          working_directory: example/android
+          command: |
+            # flutter build generates files in android/ for building the app
+            flutter build apk
+            ./gradlew app:assembleAndroidTest
+            ./gradlew app:assembleDebug -Ptarget=`pwd`/../integration_test/app_test.dart
+      - persist_to_workspace:
+          root: .
+          paths:
+            - example/build/app/outputs/apk/debug/app-debug.apk
+            - example/build/app/outputs/apk/androidTest/debug/app-debug-androidTest.apk
+
+  run-firebase-tests:
+    description: "Run integration tests for Flutter"
+    executor: gcp-cli/google
+    steps:
+      - checkout
+      - attach_workspace:
+          at: .
+      - gcp-cli/initialize:
+          gcloud-service-key: GCLOUD_SERVICE_KEY
+          google-compute-zone: GOOGLE_COMPUTE_ZONE
+          google-project-id: GOOGLE_PROJECT_ID
+      - run:
+          name: Test with Firebase Test Lab
+          command: >
+            gcloud firebase test android run --type instrumentation \
+              --app example/build/app/outputs/apk/debug/app-debug.apk \
+              --test example/build/app/outputs/apk/androidTest/debug/app-debug-androidTest.apk \
+              --timeout 2m \
+              --results-bucket cloud-test-${GOOGLE_PROJECT_ID}
+      - run:
+          name: Copy test results data
+          command: |
+            mkdir -p ~/gsutil/
+            gsutil -m cp -r -U `gsutil ls gs://cloud-test-$GOOGLE_PROJECT_ID | tail -1` ~/gsutil/ | true
+      - store_artifacts:
+          path: ~/gsutil/
+      - store_test_results:
+          path: ~/gsutil/
+
+  ios-integration-test:
+    description: "Run integration tests for Flutter"
+    macos:
+      xcode: 11.3.1
+    steps:
+      - checkout
+      - run:
+          name: Open simulator
+          command: xcrun instruments -w "iPhone 11 Pro (13.3) [" || true
+      - setup-flutter
+      - restore_cache:
+          name: Restore cocoapods specs and pods
+          key: v1-cocoapods-{{ checksum "./example/ios/Podfile.lock" }}-{{ arch }}
+      - prepare-tests
+      - build-flutter-project:
+          project: example
+      - run-flutter-integration-tests
```

Comment:
is this guaranteed to use the `iPhone 11 Pro (13.3)` or is it the device default for the executor and xcode version?

Reply:
It looks like it runs in the booted device. I upgraded xcode to 12.2 and it still uses the iPhone 11 Pro that is started couple steps before